### PR TITLE
planner: Move the Selectivity function from the stats package into cardinality package

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #46358

Problem Summary: Move the Selectivity function from the stats package into cardinality package

### What is changed and how it works?

**No logical change, just some code movement.**

Follow with https://github.com/pingcap/tidb/pull/46359

Move the Selectivity function from the stats package.

image

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/table.go [hunk 1]
```diff
@@ -63,6 +60,20 @@ const (
 	PseudoRowCount = 10000
 )
 
+var (
+	// Below functions are used to solve cycle import problem.
+	// TODO: resolve the cycle import problem in a more graceful way.
```
Comment (qw4990) [inline statistics/table.go:65]: This cycle import problem will be naturally resolved as we move more estimation functions from `stats` to `cardinality` package.

**Review by elsa0520 — APPROVED** [review]
LGTM

File: planner/cardinality/row_count_column.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package cardinality
```
Comment (hawkingrei) [inline planner/cardinality/row_count_column.go:1]: add license header.

Comment (qw4990) [reply planner/cardinality/row_count_column.go:15]: udpated.

File: statistics/handle/handletest/handle_test.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package handletest
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/planner/cardinality"
```
Comment (hawkingrei) [inline statistics/handle/handletest/handle_test.go:19]: sort the import.

Comment (qw4990) [reply statistics/handle/handletest/handle_test.go]: OK!

**Review by hawkingrei — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**